source/extensions/common/wasm/wasm.cc

```diff
@@ -1722,21 +1722,38 @@ const StreamInfo::StreamInfo* Context::getRequestStreamInfo() const {
     return &decoder_callbacks_->streamInfo();
   } else if (access_log_stream_info_) {
     return access_log_stream_info_;
+  } else if (network_read_filter_callbacks_) {
+    return &network_read_filter_callbacks_->connection().streamInfo();
+  } else if (network_write_filter_callbacks_) {
+    return &network_write_filter_callbacks_->connection().streamInfo();
   }
   return nullptr;
 }
 
+// FilterState
+StreamInfo::FilterState& Context::getFilterState() const {
+  if (encoder_callbacks_) {
+    return encoder_callbacks_->streamInfo().filterState();
+  }
+  if (decoder_callbacks_) {
+    return decoder_callbacks_->streamInfo().filterState();
+  }
+  if (network_read_filter_callbacks_) {
+    return network_read_filter_callbacks_->connection().streamInfo().filterState();
+  }
+  return network_write_filter_callbacks_->connection().streamInfo().filterState();
```

Comment:
I don't think this is correct, you have no guarantee that this isn't `nullptr`.

Also, does `network_read_filter_callbacks_->connection()` and `network_write_filter_callbacks_->connection()` point to the same connection or is one for downstream and one for upstream?

Reply:
Thanks for pointing..will fix that..

looks like they both point to the same connection.. we need different one for upstream and downstream right?